packages/paste-website/src/pages/tokens/how-to-compose-custom-ui-with-tokens.mdx

```diff
@@ -0,0 +1,137 @@
+
+## Design Primitive Components
+
+Design Primitive Components are React components that expose style properties using [Styled-System](https://www.styled-system.com). These style properties are typed React props that only expose design tokens as values. There are style properties for each of the categories that are included on the [theme object](theme-package#theme) from the [theme package](theme-package).
+
+We use these components internally to create almost everything in Paste. When used, they provide an incredible number of different combinations of design tokens that can be used to create new, unique things that will look exactly like they came directly from Paste. They also provide support for the themes that Paste offers out of the box and a free upgrade path to any new design language updates in the future. Your custom components will never be left behind.
+
+Each Design Primitive is an Emotion Styled Component that creates styles that are encapsulated to component. This way you can style your custom components with the confidence that your styles should never clash with another teams work.
+
+### Box
+
+The Box component is designed to be a block level utility, you can think of it as a super `div`. By exposing the `as` prop any box can become any valid HTML element, but obviously the sensible choices would be box or block style things. It exposes box like styling properties like background color, padding, margin, border styles, etc, to allow you to create custom block level containers. The values the styling properties take are the names of the design tokens exposed on the theme object.
+
+You can compose any number of Boxes together to create new UI components. Here's an example of 3 boxes, 2 of which are nested in another.
+
+<LivePreview scope={{Box}} language="jsx">
+  {`<Box
+  as="article"
+  backgroundColor="colorBackgroundBody"
+  padding="space60"
+>
+  <Box
+    backgroundColor="colorBackgroundSuccessLightest"
+    display="inline-block"
+    padding="space40"
+  >
+    Some child elements
+  </Box>
+  <Box
+    backgroundColor="colorBackgroundPrimaryLight"
+    display="inline-block"
+    padding="space40"
+  >
+    Some child elements
+  </Box>
+</Box>`}
+</LivePreview>
+
+
+### Text
+
+The text component is designed to be a general purpose text utilty. It too exposes the `as` prop to allow it to be transformed into any valid HTML element and comes with a number of text styling properties. These properties include font-size, text color, weight and line height and allow you to create text styles that should match the design system.
+
+<LivePreview scope={{Box, Text}} language="jsx">
+  {`<Box padding="space20">
+  <Text as="p" marginBottom="space40">Some text</Text>
+  <Text as="h1" fontSize="fontSize70" fontWeight="fontWeightMedium" lineHeight="lineHeight70" mb="space80">Custom heading</Text>
+  <Text as="ul" ml="space60">
```

Comment:
Should this be a Text or a Box?

If we're making it a Text, can we pass the fontSize to it and would the children elements receive the styling?

Reply:
So `Text` right now has a set default font-size, which matches what the system classes as the default body size. The reason being is that we font set a body size as you can't really override someone's application, and we don't actually know where this will be used so can't rely on any inheritance.

So even if I did set it on the ul, it won't currently affect the children as they set their own font-size anyway.

Reply:
Makes sense.  Is there any guidelines or issues with making a `ul` element a `Text` as opposed to a `Box` though?